spec/unit/puppet-strings/yard/handlers/ruby/function_handler_spec.rb

```diff
@@ -19,33 +19,30 @@
 
   describe 'parsing 3.x API functions' do
     describe 'parsing a function with a missing docstring' do
-      let(:source) do
-        <<-SOURCE
-Puppet::Parser::Functions.newfunction(:foo) do |*args|
-end
+      let(:source) { <<~'SOURCE' }
+        Puppet::Parser::Functions.newfunction(:foo) do |*args|
+        end
       SOURCE
-      end
 
       it 'logs a warning' do
         expect { spec_subject }.to output(%r{\[warn\]: Missing documentation for Puppet function 'foo' at \(stdin\):1\.}).to_stdout_from_any_process
       end
     end
 
     describe 'parsing a function with a doc parameter' do
-      let(:source) do
-        <<-SOURCE
-Puppet::Parser::Functions.newfunction(:foo, doc: <<-DOC
-An example 3.x function.
-@param [String] first The first parameter.
-@param second The second parameter.
-@return [Undef] Returns nothing.
-DOC
-) do |*args|
-end
+      let(:source) { <<~'SOURCE' }
+        Puppet::Parser::Functions.newfunction(:foo, doc: <<~'DOC') do |*args|
```

Comment:
Looks like I found a weird bug. Putting `) do |*args|` on the same line as the heredoc rather than after the final symbol (as it was previously) causes some number of characters to be trimmed from the docstring before YARD parses the tags (I think).

If you replace `|*args|` with `|*|` then 4 fewer characters will be trimmed.

This shows up as the `@return` tag having trimmed text, but if you put one of the `@param`s on the end, then that will have trimmed text instead. (Which is why I think it happens before it tries to look for tags.)

I’m guessing this is bug in YARD or… the Ruby parser? I’m not sure what YARD uses under the hood. I’ll try to figure out where this should be filed tomorrow.